node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,339 @@
+                                longPassword,
+                                false,
+                            ),
+                        ).resolves.toBe("OK");
+                        await client.configSet({
+                            requirepass: "",
+                        });
+                    } finally {
+                        client?.close();
+                    }
+                }, protocol);
+            });
+
+            /**
+             * Test that re-authentication fails when using wrong password.
+             */
+            it("test_replace_password_immediateAuth_wrong_password", async () => {
+                await runTest(async (client: BaseClient) => {
+                    try {
+                        await client.configSet({
+                            requirepass: NEW_PASSWORD,
+                        });
+                        await expect(
+                            client.updateConnectionPassword(
+                                WRONG_PASSWORD,
+                                true,
+                            ),
+                        ).rejects.toThrow(RequestError);
+                        await client.updateConnectionPassword(
+                            NEW_PASSWORD,
+                            true,
+                        );
+                        await client.configSet({
+                            requirepass: "",
+                        });
+                    } finally {
+                        client?.close();
+                    }
+                }, protocol);
+            });
+
+            /**
+             * Test replacing connection password with immediate re-authentication.
+             */
+            it(
+                "test_update_connection_password_with_immediateAuth",
+                async () => {
+                    await runTest(async (client: BaseClient) => {
+                        // Set server password
+                        await client.configSet({ requirepass: NEW_PASSWORD });
+
+                        // Update client password with re-auth
+                        await expect(
+                            client.updateConnectionPassword(NEW_PASSWORD, true),
+                        ).resolves.toBe("OK");
+
+                        // Verify client works with new auth
+                        await client.set("test_key", "test_value");
+                        const value = await client.get("test_key");
+                        expect(value).toEqual("test_value");
+
+                        // Clean up
+                        await client.configSet({ requirepass: "" });
+                    }, protocol);
+                },
+                TIMEOUT,
+            );
+
+            /**
+             * Test changing server password when connection is lost before password update.
+             * Verifies that the client will not be able to reach the inner core and return an error.
+             */
```

Comment:
Maybe describe this case a little better and point out why this only happens with the standalone client?  Seems like for a standalone client, the 'inner core' makes a request to redis (which fails) and thus the updateConnectionPassword throws a RequestError. 
... but I'm just guessing and it would be nice to have this explained somewhere.

Reply:
Ill describe more.